internal/mode/static/telemetry/collector.go

```diff
@@ -378,3 +402,110 @@ func collectClusterInformation(ctx context.Context, k8sClient client.Reader) (cl
 
 	return clusterInfo, nil
 }
+
+type sfContextDirective struct {
+	context   string
+	directive string
+}
+
+func collectSnippetsFilterSnippetsInfo(graphGetter GraphGetter) ([]string, []int64, error) {
+	g := graphGetter.GetLatestGraph()
+	if g == nil {
+		return nil, nil, errors.New("latest graph cannot be nil")
+	}
+
+	contextDirectiveMap := make(map[sfContextDirective]int)
+
+	for name := range g.SnippetsFilters {
+		sf := g.SnippetsFilters[name]
+		if sf == nil {
+			continue
+		}
```

Comment:
You can just do:

```
for name, sf := range g.SnippetsFilters {
```

If sf exists, can it possibly be nil?

Reply:
Ok, I think since sf is a pointer to a `graph.SnippetsFilter`, that means it could be nil. So I'll leave that check in there.